proto/cosmos/bank/v1beta1/authz.proto

```diff
@@ -19,6 +19,7 @@ message SendAuthorization {
   repeated cosmos.base.v1beta1.Coin spend_limit = 1 [
     (gogoproto.nullable)     = false,
     (amino.dont_omitempty)   = true,
+    (amino.encoding)         = "null_slice_as_empty",
```

Comment:
This field encoding is only used to replicate the existing `Coins` override at https://github.com/cosmos/cosmos-sdk/blob/be9bd7a8c1b41b115d58f4e76ee358e18a52c0af/types/coin.go#L199-L205.

When `spend_limit = null`, without this field encoder:
```
{"spend_limit":null}
```
with this field encoder (desired):
```
{"spend_limit":[]}
```

Reply:
The annotations are specifying different things.

`amino.dont_omit_empty = true` -> marshal this field even if its null
`amino.encoding = "null_slice_as_empty"` -> encode a null slice as an empty list.  

Note this is not the default behavior, other slice fields could have `amino.dont_omit_empty = true`, and 1) no special encoder on the gogo side, and 2) no `amino.encoding` either, so equivalence means leaving `amino.encoding` off of the field.

Reply:
Ah, true, I somehow thought the zero value of `[]Coin` was an empty list, not a null slice, which would have simplified things.

Coins is used everywhere, I'm slightly leaning towards convention here.

Reply:
I tried this refactor.  Removed 25 usages but there were 3 types which didn't have "null_slice_as_empty", which now marshall as `[]` where the legacy codec marshals as `null`.  This is because their type in the gogo generated message is []Coin instead of the Coins type alias.  They are:

```
cosmos.gov.v1.MsgSubmitProposal
cosmos.gov.v1.MsgDeposit
cosmos.gov.v1.MsgUpdateParams
```

So these messages would need a special case annotation on their field to _ignore_ the convention.  Which alternative is best here?

Reply:
> Another option would be set (gogoproto.castrepeated) = "github.com/cosmos/cosmos-sdk/types.Coins", but will that be consensus breaking?

Yes, and also (golang) API breaking. I'm not sure this is desirable.

> So these messages would need a special case annotation on their field to ignore the convention.

I personally think this is the worst solution.

After exploring all these solutions, I think the best one is still `null_slice_as_empty`. For new proto messages, it's fine if app developers don't include this annotation on Coins. For existing messages, it should be made clear that adding/removing this annotation is a consensus-breaking change.